src/app/features/login/login.component.ts

```diff
@@ -1,14 +1,46 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, EventEmitter, Output, ViewChild } from '@angular/core';
+
+import { APP_ROUTS, User } from 'src/app/app-model';
+import {
+  faEye,
+  faEyeSlash,
+  IconDefinition,
+} from '@fortawesome/free-solid-svg-icons';
+import { InitialUserData } from './login-model';
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.scss'],
 })
-export class LoginComponent implements OnInit {
-  constructor() {}
+export class LoginComponent {
+  @ViewChild('formLogin') formLogin: any;
+
+  isPasswordHide: boolean = true;
+  passwordIcon: IconDefinition = faEye;
+  passwordTextType: string = '';
+
+  userData: User = InitialUserData;
+
+  @Output() navigateEvent = new EventEmitter<APP_ROUTS>();
+  @Output() loginEvent = new EventEmitter<User>();
+
+  login(): void {
+    if (this.formLogin.form.status === 'VALID') {
+      this.loginEvent.emit(this.userData);
+      this.navigateEvent.emit(APP_ROUTS.COURSES);
+    }
+  }
+
+  navigateToRegistration(event: Event): void {
+    event.preventDefault();
+    this.navigateEvent.emit(APP_ROUTS.REGISTRATION);
+  }
+
+  onPasswordIconClick(): void {
```

Comment:
This method we meet twice in the code, as a good tone, we can create a separate helper class, which we implement this method as a static, and use everywhere we need like this -` classNam.ourMethodName()`.

Reply:
Moved the password switch logic to a separate class, and extend from it